controllers/monitor.go

```diff
@@ -0,0 +1,169 @@
+package controllers
+
+import (
+	"context"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	trustyaiopendatahubiov1alpha1 "github.com/trustyai-explainability/trustyai-service-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var isCentralServiceMonitorCreated = false
+
+// generateServiceMonitorSpecCentral generates the ServiceMonitor spec for central ServiceMonitor
+func generateServiceMonitorSpecCentral(deploymentNamespace string) *monitoringv1.ServiceMonitor {
```

Comment:
As far as I see the owner ref is not configured for this resource.
Given that it is created in the same namespace of the controller I would set the controller as owner.
Wdyt?

Reply:
- https://github.com/trustyai-explainability/trustyai-service-operator/issues/83